content/en/blog/_posts/2021-12-15-pod-security-admission-psa-beta.md

```diff
@@ -0,0 +1,620 @@
+kind: Pod
+metadata:
+  name: nginx-privileged
+spec:
+  containers:
+  - name: nginx
+    image: nginx
+    securityContext:
+      allowPrivilegeEscalation: true
+EOF
+```
+
+The output is similar to this:
+
+```shell
+Error from server (Forbidden): error when creating "STDIN": pods "nginx-privileged" is forbidden: violates PodSecurity "restricted:latest": allowPrivilegeEscalation != false (container "nginx" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "nginx" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "nginx" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "nginx" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
+```
+
+Now let's try the `kube-test-privileged` namespace
+
+```shell
+cat <<EOF | kubectl -n kube-test-privileged apply -f -
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nginx-privileged
+spec:
+  containers:
+  - name: nginx
+    image: nginx
+    securityContext:
+      allowPrivilegeEscalation: true
+EOF
+```
+
+The output is similar to this:
+
+```shell
+pod/nginx-privileged created
+```
+
+We can run `kubectl -n kube-test-privileged get pods` to verify it is running.
+
+**Baseline level and workload**
+
+The baseline pod demonstrates sensible defaults but does not allow privilege escalation, so the pod above would fail here too ([more info](https://kubernetes.io/docs/concepts/security/pod-security-standards/#baseline)). Baseline accepts null / no values for unset keys.
+
+First, let's try to deploy it in the restricted namespace:
+
+```shell
+cat <<EOF | kubectl -n kube-test-restricted apply -f -
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nginx-baseline
+spec:
+  containers:
+  - name: nginx
+    image: nginx
+    securityContext:
+      allowPrivilegeEscalation: false
+      capabilities:
+        add:
+          - NET_BIND_SERVICE
+          - CHOWN
+EOF
+```
+
+The output is similar to this:
+
+```shell
+Error from server (Forbidden): error when creating "STDIN": pods "nginx-baseline" is forbidden: violates PodSecurity "restricted:latest": unrestricted capabilities (container "nginx" must set securityContext.capabilities.drop=["ALL"]; container "nginx" must not include "CHOWN" in securityContext.capabilities.add), runAsNonRoot != true (pod or container "nginx" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "nginx" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
+```
+
+Now let's try the `kube-test-baseline` namespace
+
+```shell
+cat <<EOF | kubectl -n kube-test-baseline apply -f -
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nginx-baseline
+spec:
+  containers:
+  - name: nginx
+    image: nginx
+    securityContext:
+      allowPrivilegeEscalation: false
+      capabilities:
+        add:
+          - NET_BIND_SERVICE
+          - CHOWN
+EOF
+```
+
+The output is similar to this:
```

Comment:
it's easy to miss that `pod/nginx-baseline created`. Consider highlighting it here.
```suggestion
The output is similar to the following. Note that the warnings match the error message from the test above, but the pod is still successfully created.
```